modules/xmpp/StropheLastSuccess.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Attaches to the {@link Strophe.Connection.rawInput} which is called whenever any data is received from the server.
+ */
+export default class LastRequestTracker {
+    /**
+     * Initializes new instance.
+     */
+    constructor() {
+        this._lastSuccess = null;
+    }
+
+    /**
+     * Starts tracking requests on the given connection.
+     *
+     * @param {Object} stropheConnection - Strophe connection instance.
+     */
+    startTracking(stropheConnection) {
+        const originalRawInput = stropheConnection.rawInput;
+
+        stropheConnection.rawInput = function(...args) {
+            this._lastSuccess = new Date().getTime();
```

Comment:
use performane.now() ?

Reply:
do you know if that's available on both web and mobile RN?

Reply:
We polyfill it with Date.now() IIRC because of callstats, but please double check

Reply:
well  I dont see point in doing that to be honest

Reply:
having it in the "since epoch" format makes more sense in this context for me